models/userStatus.js

```diff
@@ -325,6 +325,83 @@ const updateStatusOnTaskCompletion = async (userId) => {
   }
 };
 
+const massUpdateIdleUsers = async (users) => {
+  const currentTimeStamp = new Date().getTime();
+  const batch = firestore.batch();
+  let usersWithStatusUnchanged = 0;
+
+  await Promise.all(
+    users.map(async (userId) => {
+      let latestStatusData;
+      try {
+        latestStatusData = await getUserStatus(userId);
+      } catch (error) {
+        usersWithStatusUnchanged++;
+        return batch;
+      }
+      const { id, userStatusExists, data } = latestStatusData;
+
+      if (!userStatusExists || !data?.currentStatus) {
```

Comment:
I am confused with this if condition...can you give me brief idea

Reply:
`! userStatusExists or !data?.currentStatus` implies that userStatus doesn't exist. Ideally now the user status gets created on onboarding for a new user but earlier it didn't exist so it will get created by default.